### ISSUE TITLE
Clean up on the redundant import & comment

### ISSUE THREAD

**Issue by dcrowTMH** [errchh/DynPrompt-1s#1]
@errchh This is Ming. Hope you are doing fine after the pycon.
I finally got back my original git account. I think there's some redundant or duplicated import in the ui.py.
See if you would like me to help to clean. We can try to create another branch & do a pull request.
So that you can take a look on my modification first before you merge it to your main branch.

**Comment by errchh** [comment]
Hi Ming, brilliant, sure. Thanks for your dedication. -- Please proceed.

**Comment by errchh** [comment]
Updates —

Hi Ming, I have changed the directory. Moved ui.py to /src. 
And kept requirements.txt in root dir. 

To do — 

@errchh create a Dockerfile for users have Ollama already installed. 
@dcrowTMH create a Dockerfile for users do not have Ollama installed. (Ollama install in venv with pip install ollama?)

Next — 

I have a ML class starting this Thu. The class will take 2 months. I’ll be back and active by the end of Jan. 

By then, see if there is anything to further develop for this project, or take on new project which we could have exposures to other things to try and learn. 

=====
Reference on Dockerfile — 

Having two Dockerfiles in your project can be useful for different purposes, such as one for development and one for production. You can organize them by naming them appropriately and updating your build commands accordingly. 

Here's an example of how you might structure your files and directories:

```plaintext
/your_project_root
│
├── Dockerfile.dev
├── Dockerfile.prod
├── requirements.txt
└── src
    └── main.py
```

In this structure:
- `Dockerfile.dev` is for your development environment.
- `Dockerfile.prod` is for your production environment.
- Both `Dockerfile.dev` and `Dockerfile.prod` are located in the root directory along with `requirements.txt`.
- The `src` directory contains the `main.py` file.

To build and run a container using a specific Dockerfile, you would specify the filename with the `-f` option in your `docker build` command. For example:

- For development:
  ```bash
  docker build -f Dockerfile.dev -t myapp:dev .
  ```

- For production:
  ```bash
  docker build -f Dockerfile.prod -t myapp:prod .
  ```

**Comment by dcrowTMH** [comment]
> Updates —
> 
> Hi Ming, I have changed the directory. Moved ui.py to /src. And kept requirements.txt in root dir.
> 
> To do —
> 
> @errchh create a Dockerfile for users have Ollama already installed. @dcrowTMH create a Dockerfile for users do not have Ollama installed. (Ollama install in venv with pip install ollama?)
> 
> Next —
> 
> I have a ML class starting this Thu. The class will take 2 months. I’ll be back and active by the end of Jan.
> 
> By then, see if there is anything to further develop for this project, or take on new project which we could have exposures to other things to try and learn.
> 
> ===== Reference on Dockerfile —
> 
> Having two Dockerfiles in your project can be useful for different purposes, such as one for development and one for production. You can organize them by naming them appropriately and updating your build commands accordingly.
> 
> Here's an example of how you might structure your files and directories:
> 
> ```
> /your_project_root
> │
> ├── Dockerfile.dev
> ├── Dockerfile.prod
> ├── requirements.txt
> └── src
>     └── main.py
> ```
> 
> In this structure:
> 
> * `Dockerfile.dev` is for your development environment.
> * `Dockerfile.prod` is for your production environment.
> * Both `Dockerfile.dev` and `Dockerfile.prod` are located in the root directory along with `requirements.txt`.
> * The `src` directory contains the `main.py` file.
> 
> To build and run a container using a specific Dockerfile, you would specify the filename with the `-f` option in your `docker build` command. For example:
> 
> * For development:
>   ```shell
>   docker build -f Dockerfile.dev -t myapp:dev .
>   ```
> * For production:
>   ```shell
>   docker build -f Dockerfile.prod -t myapp:prod .
>   ```

That's cool! Let me have a look on it and get back to you.

**Comment by dcrowTMH** [comment]
https://github.com/valiantlynx/ollama-docker

ref

**Comment by errchh** [comment]
Brilliant Ming. Let me read later to understand and learn this. 

One thing as CUDA package is for nvidia GPU only. And the same for ML libraries, those involving GPU gets super complicated.

**Comment by dcrowTMH** [comment]
Yup. I think we can start on the simple one first as CUDA one may need further coding.
I am trying to construct the Dockerfile & test it on my Window.

I think we can simply using the sys.argv to handle the command on both docker & streamlit in the dockerfile.

**Comment by errchh** [comment]
Yes, the simple one is much more a real world use case. 

And btw, are you coming to OSHK this Fri?

**Comment by dcrowTMH** [comment]
Yup. I will be joining as I am not free last time.
